packages/trip-form/src/ModeSelector/index.js

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { modeSelectorOptionsType } from "@opentripplanner/core-utils/lib/types";
+
+import { MainModeRow, SecondaryModeRow, TertiaryModeRow } from "./styled";
+import ModeButton from "../ModeButton";
+
+/**
+ * ModeSelector is the control container where the OTP user selects
+ * the transportation modes for a trip query, e.g. transit+bike, walk, micromobility...
+ */
+const ModeSelector = props => {
+  const { className, modes, onChange, style } = props;
+  const { primary, secondary, tertiary } = modes || {
+    primary: null,
+    secondary: null,
+    tertiary: null
+  };
+
+  const handleClick = option => {
+    if (!option.selected && typeof onChange === "function") {
+      onChange(option.id);
+    }
+  };
+
+  const makeButton = option => (
+    <ModeButton
+      /* If key is missing, a warning in React Dev Tools is triggered. */
```

Comment:
This comment probably isn't needed. That's a relatively well known prop/warning.